sys/vm/vm_cheri_revoke.c

```diff
@@ -478,6 +505,10 @@ vm_cheri_revoke_object_at(const struct vm_cheri_revoke_cookie *crc, int flags,
 	}
 
 visit_ro:
+	/*
+	 * XXXMJ I think the page needs to be busy for PGA_WRITEABLE to be
+	 * stable.
+	 */
```

Comment:
Oh that's fair.  Should this be doing something like the dance suggested in your added comment above, too, i.e.
```
if (pmap_page_is_write_mapped(m)) {
  vm_page_busy(m);
  if (pmap_page_is_write_mapped(m)) {
    // ...
```

Reply:
I suspect so, yes. In particular, vm_cheri_revoke_visit_(rw|ro)() both xbusy the page, so it seems that that xbusy could be lifted out.